### PR TITLE
Emit event when component has been removed

### PR DESCRIPTION
**Pull request by mayognaise**
This PR is for this issue:
https://github.com/aframevr/aframe/issues/1431

Additionally it's been attached `name` of component

### CODE REVIEW

File: src/core/a-entity.js [hunk 1]
```diff
@@ -312,6 +312,7 @@ var proto = Object.create(ANode.prototype, {
       pauseComponent(component, this.sceneEl);
       component.remove();
       delete this.components[name];
+      this.emit('componentremoved');
```
Comment (donmccurdy) [inline src/core/a-entity.js]: Might be helpful to attach the name of the component, as well –

```javascript
this.emit('componentremoved', {name: 'name'});
```

Comment (mayognaise) [inline src/core/a-entity.js]: @donmccurdy great idea let me add it.

**Comment by ngokevin** [comment]
Can you write a quick unit test for this in `a-entity.test.js`? Thanks!

**Comment by mayognaise** [comment]
@ngokevin hmm actually the test hasn't been passed.. the event is not called :/ do you think it's because it's already removed?

**Comment by ngokevin** [comment]
Don't need to create a Mesh, you should set a random component, use `removeAttribute` on it, and check the event detail has the component name

**Comment by mayognaise** [comment]
@ngokevin using `entityFactory();`?

**Comment by ngokevin** [comment]
In the test suite, there should already be a `setup` that calls `entityFactory` for you.

`removeAttribute` detaches components, which is what you're testing here. `removeChild` would detach all components I guess, but it's simpler to test with `removeAttribute`.

**Comment by ngokevin** [comment]
ex:

```js
  test('emits `componentremoved` event when element itself has been removed', function (done) {
    var el = this.el;
    el.setAttribute('geometry');
    el.addEventListener('componentremoved', function (evt) {
      assert.equal(evt.detail.name, 'geometry');
      done();
    });
    el.removeAttribute('geometry');
  });
```

**Comment by mayognaise** [comment]
ohh, oh i see 
so this doesn’t mean when the `a-entity` has been removed from dom, correct?

**Comment by ngokevin** [comment]
yes just testing component detach

**Comment by mayognaise** [comment]
@ngokevin i see. `componentremoved` hasn't been fired.. 😭

**Comment by ngokevin** [comment]
`el.setAttribute('geometry', '');` fixed this line

**Comment by ngokevin** [comment]
you might want something for detachedCallback, but this PR is ok too

**Comment by ngokevin** [comment]
I'll check this PR out later and try to add a proper test.

**Comment by mayognaise** [comment]
@ngokevin the test has been passed 🎉

File: src/core/a-entity.js [hunk 2]
```diff
@@ -312,6 +312,7 @@ var proto = Object.create(ANode.prototype, {
       pauseComponent(component, this.sceneEl);
       component.remove();
       delete this.components[name];
+      this.emit('componentremoved', { name: name });
```
Comment (dmarcos) [inline src/core/a-entity.js]: should we pass the entity as well?

Comment (dmarcos) [inline src/core/a-entity.js]: not sure if we should pass it. it would be done this way:
```this.emit('componentremoved', { el: this, name: name });```

Comment (mayognaise) [inline src/core/a-entity.js]: @dmarcos sure let me add `el`

Comment (donmccurdy) [inline src/core/a-entity.js]: would it already be available as `event.target`? that seems like enough if so, otherwise maybe helpful to add it.

Comment (mayognaise) [inline src/core/a-entity.js]: ah true, let me check

Comment (mayognaise) [inline src/core/a-entity.js]: This has passed. so maybe dont we need to add it?
```js
  test('emits `componentremoved` event when element itself has been removed', function (done) {
    var el = this.el;
    el.setAttribute('geometry', 'primitive:plane');
    el.addEventListener('componentremoved', function (event) {
      assert.equal(event.target, el);
      assert.equal(event.detail.name, 'geometry');
      done();
    });
    el.removeAttribute('geometry');
  });
```

**Comment by dmarcos** [comment]
Nice work. @mayognaise thanks!

**Pull request merged**